pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ConstructFromTextFunction.java

```diff
@@ -0,0 +1,78 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.geospatial.transform.function;
+
+import com.google.common.base.Preconditions;
+import org.apache.pinot.core.common.DataSource;
+import org.apache.pinot.core.geospatial.serde.GeometrySerializer;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.core.operator.transform.function.BaseTransformFunction;
+import org.apache.pinot.core.operator.transform.function.TransformFunction;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKTReader;
+
+import java.util.List;
+import java.util.Map;
+
+public abstract class ConstructFromTextFunction extends BaseTransformFunction {
+    private TransformFunction _transformFunction;
+    private byte[][] _results;
+    private WKTReader _reader;
+
+    @Override
+    public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+        Preconditions
+                .checkArgument(arguments.size() == 1, "Exactly 1 argument is required for transform function: %s",
+                        getName());
+        TransformFunction transformFunction = arguments.get(0);
+        Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
+                "The argument must be single-valued for transform function: %s", getName());
+        _transformFunction = transformFunction;
+        _reader = new WKTReader(getGeometryFactory());
+    }
+
+    abstract protected GeometryFactory getGeometryFactory();
+
+    @Override
+    public TransformResultMetadata getResultMetadata() {
+        return BYTES_SV_NO_DICTIONARY_METADATA;
+    }
+
+    @Override
+    public byte[][] transformToBytesValuesSV(ProjectionBlock projectionBlock) {
+        if (_results == null) {
+            _results = new byte[DocIdSetPlanNode.MAX_DOC_PER_CALL][];
+        }
+        String[] argumentValues = _transformFunction.transformToStringValuesSV(projectionBlock);
+        int length = projectionBlock.getNumDocs();
+        for (int i = 0; i < length; i++) {
+            try {
+                Geometry geometry = _reader.read(argumentValues[i]);
+                _results[i] = GeometrySerializer.serialize(geometry);
+            } catch (ParseException e) {
+                throw new RuntimeException(String.format("Failed to parse geometry from String %s", argumentValues[i]));
```

Comment:
Utils.rethrow will preserve the original exception.

Reply:
Good to see this util